listing-bundle/src/Resources/contao/modules/ModuleListing.php

```diff
@@ -359,9 +359,12 @@ protected function compile()
 	 */
 	protected function listSingleRecord($id)
 	{
+		/** @var PageModel $objPage */
+		global $objPage;
+
 		$this->Template = new FrontendTemplate($this->list_info_layout ?: 'info_default');
 		$this->Template->record = array();
-		$this->Template->referer = 'javascript:history.go(-1)';
+		$this->Template->referer = $objPage->getFrontendUrl();
```

Comment:
Not sure about this change as there could be other GET parameters besides `/show/42` that need to be preserved (e.g. pagination). @contao/developers Any idea how to improve this?

Reply:
That's the case with news and events as well. Your news list could be paginated and you might have been on page 3 when you clicked on a link to a news detail page. The back link would then show the news list on page 1 again. I don't think this can be solved without using the _actual_ the visitor came from.

We could check the referrer URL and see if it matches the overview URL and then use the whole referrer URL instead or something?

Reply:
> We could check the referrer URL and see if it matches the overview URL and then use the whole referrer URL instead or something?

I’m not a a fan of that, IMO the link should always point to the same URL, no matter from where you came. (Also important to make it cachable)

Reply:
As briefly discussed with @ausi, we will leave the `javascript:history.go(-1)` here for the time being. We can still fix this in Contao 5 by either storing the filter settings in the session storage of the browser or by reworking the module entirely.

Reply:
Why do we need this special handling just for the listing module? What's the difference to a paginated news list for example?

Reply:
The news module takes you back to the canonical URL, which is fine. The listing module, however, has a search field and a "Results per page" dropdown, which should not be reset when going back.

<img width="729" alt="" src="https://user-images.githubusercontent.com/1192057/148380166-6b322fd1-dcb6-45cd-ba7b-e22daf85cd51.png">

Reply:
> which should not be reset when going back.

Why not? Why is it okay to remove the query parameter for a paginated news list, but not for a filtered listing module? I do not see the difference.